frontend/packages/component-library/src/video/Video.tsx

```diff
@@ -0,0 +1,135 @@
+ * * (✔) has storybook, covered with stories and documentation;
+ * * (✔) has tests: test every prop, every state and every interaction that's js related;
+ * * (see ./__tests__/Video.test.tsx)
+ * * (✔) passes accessibility checks;
+ *
+ * ###  Status: ```Ready for dev```
+ *
+ * Design System: Video Component.
+ * This component is used to display a video from YouTube with a fallback HTML video player,
+ * and the option to download it, if an externally hosted fallback is provided.
+ * The video can also be displayed with a caption and works with responsive screen sizes.
+ */
+
+export const Video: React.FC<VideoProps> = ({
+  youTubeId,
+  videoTitle,
+  videoFallback,
+  showCaption,
+  downloadLabel,
+  errorHeading,
+  errorBody,
+  className,
+  ...HTMLAttributes
+}: VideoProps) => {
+  const [isYouTubeBlocked, setIsYouTubeBlocked] = useState(false);
+  const posterThumbnail = `//i.ytimg.com/vi/${youTubeId}/mqdefault.jpg`;
+
+  // Check to see if YouTube is blocked.
+  // If it is, we'll use the fallback video player.
+  useEffect(() => {
+    const checkYouTubeUrls = async () => {
+      const [isYouTubeBlocked, isYouTubeNoCookieBlocked] = await Promise.all([
+        checkIfURLIsBlocked('https://www.youtube.com/favicon.ico'),
+        checkIfURLIsBlocked('https://www.youtube-nocookie.com/favicon.ico'),
+      ]);
+
+      setIsYouTubeBlocked(isYouTubeBlocked || isYouTubeNoCookieBlocked);
+    };
+
+    checkYouTubeUrls();
+  }, []);
+
+  return (
+    <figure className={moduleStyles.videoComponentContainer}>
+      <div className={moduleStyles.videoWrapper}>
+        {isYouTubeBlocked ? (
+          videoFallback ? (
+            // Disabling this eslint rule since we don't support captions on all of our videos.
+            // eslint-disable-next-line jsx-a11y/media-has-caption
+            <video
+              className={classNames(className)}
+              title={videoTitle}
+              poster={posterThumbnail}
+              src={videoFallback}
+              controls
+              {...HTMLAttributes}
+            />
+          ) : (
+            <div className={classNames(moduleStyles.errorPlaceholder)}>
+              <FontAwesomeV6Icon
+                iconName="exclamation-circle"
+                iconStyle="solid"
+              />
+              <BodyTwoText>
+                <StrongText>{errorHeading || 'Video unavailable'}</StrongText>
+              </BodyTwoText>
+              <BodyThreeText>
+                {errorBody || 'This video is blocked on your network.'}
+              </BodyThreeText>
+            </div>
+          )
+        ) : (
+          <iframe
+            className={classNames(className)}
+            src={`https://www.youtube-nocookie.com/embed/${youTubeId}`}
+            title={videoTitle}
+            allowFullScreen
+            {...HTMLAttributes}
+          />
+        )}
+      </div>
+      <div className={moduleStyles.footer}>
+        {showCaption && (
```

Comment:
Non blocking, just for context: is it possible that we might need to add caption to `<video>` element later as well?

Reply:
This isn't dependent on the `<iframe>` or `<video>`, it's this caption down here:
<img width="985" alt="Screenshot 2025-01-30 at 10 27 57 AM" src="https://github.com/user-attachments/assets/d6c891b9-07c8-4153-8457-e7ab3217a43b" />